lib/instrumentation/modules/http.js

```diff
@@ -27,7 +27,7 @@ module.exports = function (http, agent) {
       var trans = agent._instrumentation.currentTransaction
 
       if (trans) {
-        trans.result = this.statusCode
+        trans.result = 'HTTP ' + this.statusCode.toString()[0] + 'xx'
```

Comment:
Can statusCode be `undefined` and therefore fail?

Reply:
It shouldn't be possible to get to that state. Node.js would normally throw if you set an invalid status code, e.g:

```
_http_server.js:188
    throw new RangeError(`Invalid status code: ${originalStatusCode}`);
    ^

RangeError: Invalid status code: undefined
    at ServerResponse.writeHead [as writeHeader] (_http_server.js:188:11)
    at ServerResponse.wrappedWriteHead [as writeHead] (/Users/watson/code/node_modules/elastic-apm/lib/instrumentation/modules/http.js:25:29)
    at ServerResponse._implicitHeader (_http_server.js:179:8)
    at ServerResponse.end (_http_outgoing.js:752:10)
    at Server.onRequest (/Users/watson/code/node_modules/elastic-apm/test/instrumentation/modules/http/basic.js:110:7)
    at emitTwo (events.js:125:13)
    at Server.emit (events.js:213:7)
    at Server.emit (/Users/watson/code/node_modules/elastic-apm/lib/instrumentation/http-shared.js:54:19)
    at parserOnIncoming (_http_server.js:602:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
```

But I guess it doesn't hurt to guard against it in case the behavior of Node.js changes in the future. I'll update the PR.

Reply:
Updated... in case `this.statusCode` is `undefined`, it will produce the result `HTTP undefinedxx`. Not pretty, but we're talking about a state that's technically not possible to end up with in Node.js today. So I think it's fair to not add too much logic to handle this - as long as we're not throwing.